exercises/plot_dets_filesize.py

```diff
@@ -0,0 +1,93 @@
+from databroker import Broker
+import pandas as pd
+
+import os
+
+import datetime
+import time
+from time import mktime
+
+from eiger_io.fs_handler import EigerHandler
+from databroker.assets.handlers import AreaDetectorTiffHandler
+
+
+def find_keys(hdrs, db):
+    '''
+        This function searches for keys that are stored via filestore in a
+        database, and gathers the SPEC id's from them.
+    '''
+    FILESTORE_KEY = "FILESTORE:"
+    keys_dict = dict()
+    for hdr in hdrs:
+        for stream_name in hdr.stream_names:
+            events = hdr.events(stream_name=stream_name)
+            events = iter(events)
+            while True:
+                try:
+                    event = next(events)
+                    if "filled" in event:
+                        # there are keys that may not be filled
+                        for key, val in event['filled'].items():
+                            if key not in keys_dict and not val:
+                                # get the datum
+                                if key in event['data']:
+                                    datum_id = event['data'][key]
+                                    resource = db.reg.resource_given_datum_id(datum_id)
+                                    keys_dict[key] = resource['spec']
```

Comment:
right here is where you can start trying to look at file size information

```py
datum_gen = db.reg.datum_gen_given_resource(resource)
datum_kwargs_list = [datum['datum_kwargs'] for datum in datum_gen]
fh = db.reg.get_spec_handler(resource)
file_lists = fh.get_file_list(datum_kwargs)
file_sizes = [....]
```

Reply:
So are we able to reuse the code from finding the file sizes for chx from 2015-01-01 to 2018-12-31, is it a similar approach?

Reply:
Mhm never mind that question, I'm just trying to think how to do this

Reply:
Ah I think I got it, thanks!!!!!!!